internal/provision/devices.go

```diff
@@ -23,19 +28,56 @@ import (
 	"github.com/edgexfoundry/device-sdk-go/v2/internal/container"
 )
 
-func LoadDevices(deviceList []common.DeviceConfig, dic *di.Container) errors.EdgeX {
+func LoadDevices(path string, dic *di.Container) errors.EdgeX {
+	if path == "" {
+		return nil
+	}
 	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
-	serviceName := container.DeviceServiceFrom(dic.Get).Name
+
+	absPath, err := filepath.Abs(path)
+	if err != nil {
+		return errors.NewCommonEdgeX(errors.KindServerError, "failed to create absolute path", err)
+	}
+
+	fileInfo, err := ioutil.ReadDir(absPath)
+	if err != nil {
+		return errors.NewCommonEdgeX(errors.KindServerError, "failed to read directory", err)
+	}
+
 	var addDevicesReq []requests.AddDeviceRequest
+	serviceName := container.DeviceServiceFrom(dic.Get).Name
+	lc.Debugf("Loading pre-defined devices from %s", absPath)
```

Comment:
Should this be `Info` since it happens once a start-up?

Reply:
sounds reasonable, updated here and below.